app/utils/types.ts

```diff
@@ -0,0 +1,6 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+export function secureGetFromRecord<T>(v: Record<string, T> | undefined, key: string) {
+    return typeof v === 'object' && v && Object.prototype.hasOwnProperty.call(v, key) ? v[key] : undefined;
```

Comment:
This can also be changed to `Object.hasOwn` instead of `Object.prototype.hasOwnProperty`

Reply:
Not sure about the reasons, but it seemed that the linter was complaining when I tried to use `hasOwn`. Maybe is not implemented in the version used by React Native. Didn't look much deeper :shrug: